raw.go

```diff
@@ -31,3 +31,32 @@ type RawSpan struct {
 	// The span's associated baggage.
 	Baggage map[string]string // initialized on first use
 }
+
+// SpanDetail summarizes the identifying fields of a Span, for
+// diagnostic and implementation-specific purposes.
+type SpanDetail struct {
```

Comment:
why not define this as an interface and let basictracer's Span implement it? Then you don't need to copy anything, just convert. And the `GetSpanDetail` becomes unnecessary, users can simply do a checked cast to this new interface.